lib/modules/versioning/types.ts

```diff
@@ -10,26 +10,88 @@ export interface NewValueConfig {
 }
 export interface VersioningApi {
   // validation
+
+  /**
+   * Check whether the `version` is compatible with the `current` version range
+   * constraint.
+   */
   isCompatible(version: string, current?: string): boolean;
+
+  /**
+   * Check whether the `version` constraint is pinned, i.e. it only allows a
+   * single specific version.
+   */
   isSingleVersion(version: string): boolean;
+
+  /**
+   * Check whether the `version` is considered to be "stable".
+   *
+   * Example: in SemVer the version must not have a pre-release marker.
+   */
   isStable(version: string): boolean;
+
+  /**
+   * Check whether the `input` is a valid version range constraint.
+   */
   isValid(input: string): boolean;
+
+  /**
+   * Check whether the `input` is a valid version string.
```

Comment:
@rarkins i don't remember the difference to the `isValid` function. on most versionings it's same function.

Reply:
isValid means it can be a constraint or a version, isVersion means version only

Reply:
what's the difference between `isVersion` and `isSingleVersion` ? aren't they the same then?